src/main/java/org/mvel2/ast/ASTNode.java

```diff
@@ -138,7 +138,7 @@ private Object optimize(Object ctx, Object thisValue, VariableResolverFactory fa
     AccessorOptimizer optimizer;
     Object retVal = null;
 
-    if ((fields & NOJIT) != 0 || factory != null && factory.isResolveable(nameCache)) {
+    if ((fields & NOJIT) != 0 || factory != null && nameCache !=null && factory.isResolveable(nameCache)) {
```

Comment:
I believe this fix is not entirely correct. In my opinion you shouldn't do a null check, but accessing the property name using the getName() method as it follows

`if ((fields & NOJIT) != 0 || factory != null && factory.isResolveable(getName())) {`

@gmunozfe Please let me know if you want to update your PR or you want me to push the fix that I'm suggesting on master.

Reply:
@mariofusco thanks for your review, I've fixed it according to it.
With this strategy, then, the custom resolver will receive the variable name with the deep property ($.firstName) though it won't be able to resolve it, returning false, no?
Tests are passing, so jBPM will work when upgrading to the new mvel version with this fix.